Source/RuntimeFilesDownloader/Private/RuntimeChunkDownloader.cpp

```diff
@@ -50,21 +50,26 @@ TFuture<FRuntimeChunkDownloaderResult> FRuntimeChunkDownloader::DownloadFile(con
 				if (!SharedThis.IsValid())
 					{
 					UE_LOG(LogRuntimeFilesDownloader, Warning, TEXT("Failed to download file chunk from %s: downloader has been destroyed"), *URL);
-					PromisePtr->SetValue(FRuntimeChunkDownloaderResult{EDownloadToMemoryResult::DownloadFailed, TArray64<uint8>()});
+					PromisePtr->SetValue(FRuntimeChunkDownloaderResult{EDownloadToMemoryResult::DownloadFailed, {}, {}});
 					return;
 				}
 
 				if (SharedThis->bCanceled)
 				{
 					UE_LOG(LogRuntimeFilesDownloader, Warning, TEXT("Canceled file chunk download from %s"), *URL);
-					PromisePtr->SetValue(FRuntimeChunkDownloaderResult{EDownloadToMemoryResult::Cancelled, TArray64<uint8>()});
+					PromisePtr->SetValue(FRuntimeChunkDownloaderResult{EDownloadToMemoryResult::Cancelled, {}, {}});
 					return;
 				}
 
 				PromisePtr->SetValue(FRuntimeChunkDownloaderResult{Result.Result, MoveTemp(Result.Data)});
 			});
 		};
-		
+
+		if (ContentSize == -304)
```

Comment:
Double checking this is meant to be a negative 304, if so whhyyy?

Reply:
yeah - we somehow have to pass this info through, and I didn't feel like changing the promise return parameters.

essentially, 0 and all negative values already meant "some error" - now I'm just making -304 the "not actually an error, just means we can return NotModified"

```suggestion
		// -304 is used by GetContentSize to signal that the HEAD request returned a "304 Not Modified" instead of a size.
		if (ContentSize == -304)
```